### ISSUE TITLE
npm URL throws: `error: [ERR_MODULE_NOT_FOUND] Cannot find module A imported from B`

### ISSUE THREAD

**Issue by GJZwiers** [denoland/deno#15603]
This happens with `fs-extra`:

```js
import fsx from 'npm:fs-extra@10.1.0';
```

```
deno run --unstable -A main.js

error: [ERR_MODULE_NOT_FOUND] Cannot find module "C:\Users\GJZwiers\AppData\Local\deno\npm\registry.npmjs.org\fs-extra\10.1.0\lib\./copy.js" imported from "C:\Users\GJZwiers\AppData\Local\deno\npm\registry.npmjs.org\fs-extra\10.1.0\lib\index.js"
```

```
deno --version

deno 1.25.0 (release, x86_64-pc-windows-msvc)
v8 10.6.194.5
typescript 4.7.4
```

**Comment by GJZwiers** [comment]
The same error also happens with the AWS SDK v3:

```ts
import { Lambda } from "npm:@aws-sdk/client-lambda";
```

```
Download https://registry.npmjs.org/@aws-sdk/client-lambda
Download https://registry.npmjs.org/@aws-sdk/client-lambda/-/client-lambda-3.154.0.tgz
error: [ERR_MODULE_NOT_FOUND] Cannot find module "C:\Users\GJZwiers\AppData\Local\deno\npm\registry.npmjs.org\@aws-sdk\client-lambda\3.154.0\dist-cjs\./commands.js" imported from "C:\Users\GJZwiers\AppData\Local\deno\npm\registry.npmjs.org\@aws-sdk\client-lambda\3.154.0\dist-cjs\index.js"
```

**Comment by bartlomieju** [comment]
Fixed in #15697
